File: Session-3/DDD-Sample-codes/rtgs-return-management-springboot/src/main/java/com/bank/rtgs/returnmanagement/infrastructure/payload/SimpleReturnPayloadGenerator.java
package com.bank.rtgs.returnmanagement.infrastructure.payload;

import com.bank.rtgs.returnmanagement.domain.model.RtgsReturn;
import com.bank.rtgs.returnmanagement.domain.ports.ReturnPayloadGenerator;
import org.springframework.stereotype.Component;

@Component
public class SimpleReturnPayloadGenerator implements ReturnPayloadGenerator {
    @Override
    public String generate(RtgsReturn rtgsReturn) {
        return "<RtgsReturn>" +
                "<ReturnId>" + rtgsReturn.getId() + "</ReturnId>" +
                "<OriginalTransactionReference>" + rtgsReturn.getOriginalTransactionReference().value() + "</OriginalTransactionReference>" +
                "<ReasonCode>" + rtgsReturn.getReason().code() + "</ReasonCode>" +
                "<ReasonDescription>" + rtgsReturn.getReason().description() + "</ReasonDescription>" +
                "</RtgsReturn>";
    }
}

File: Session-3/DDD-Sample-codes/rtgs-return-management-dotnet10/Infrastructure/Payload/SimpleReturnPayloadGenerator.cs
using Rtgs.ReturnManagement.Api.Domain.Model;
using Rtgs.ReturnManagement.Api.Domain.Ports;

namespace Rtgs.ReturnManagement.Api.Infrastructure.Payload;

public sealed class SimpleReturnPayloadGenerator : IReturnPayloadGenerator
{
    public string Generate(RtgsReturn rtgsReturn) =>
        "<RtgsReturn>" +
        "<ReturnId>" + rtgsReturn.Id + "</ReturnId>" +
        "<OriginalTransactionReference>" + rtgsReturn.OriginalTransactionReference.Value + "</OriginalTransactionReference>" +
        "<ReasonCode>" + rtgsReturn.Reason.Code + "</ReasonCode>" +
        "<ReasonDescription>" + rtgsReturn.Reason.Description + "</ReasonDescription>" +
        "</RtgsReturn>";
}
